contracts/scripts/interactions/InteractAlphaSparse.s.sol

```diff
@@ -0,0 +1,175 @@
+      jsonData = vm.readFile(path);
+    }
+  }
+
+  /**
+   * @notice Decodes diamond and facet data from a JSON file output by the bytecode-diff script
+   * @dev Reads the JSON file specified by the DEFAULT_JSON_FILE constant
+   *      and parses it to extract information about updated diamonds and their facets
+   * @return An array of DiamondFacets structs containing the decoded information
+   */
+
+  function decodeDiamondsFromJSON()
+    internal
+    view
+    returns (DiamondFacets[] memory)
+  {
+    uint256 updatedDiamondLen = abi.decode(
+      vm.parseJson(jsonData, ".numUpdated"),
+      (uint256)
+    );
+    DiamondFacets[] memory diamonds = new DiamondFacets[](updatedDiamondLen);
+
+    for (uint256 i = 0; i < updatedDiamondLen; i++) {
+      bytes memory diamondData = vm.parseJson(
+        jsonData,
+        string.concat(".updated[", vm.toString(i), "]")
+      );
+
+      // Decode diamond name and number of facets
+      (string memory diamondName, uint256 numFacets) = abi.decode(
+        diamondData,
+        (string, uint256)
+      );
+
+      // Create DiamondFacets struct with fixed-size facets array
+      diamonds[i] = DiamondFacets({
+        diamond: diamondName,
+        numFacets: numFacets,
+        facets: new FacetData[](numFacets)
+      });
+
+      // Decode facets one by one
+      for (uint256 j = 0; j < numFacets; j++) {
+        bytes memory facetData = vm.parseJson(
+          jsonData,
+          string.concat(
+            ".updated[",
+            vm.toString(i),
+            "].facets[",
+            vm.toString(j),
+            "]"
+          )
+        );
+        diamonds[i].facets[j] = abi.decode(facetData, (FacetData));
+      }
+    }
+
+    return diamonds;
+  }
+
+  function __interact(address deployer) internal override {
+    vm.setEnv("OVERRIDE_DEPLOYMENTS", "1");
+
+    readJSON(DEFAULT_JSON_FILE);
+
+    DiamondFacets[] memory diamonds = decodeDiamondsFromJSON();
+
+    // Iterate over diamonds array and process each diamond
+    for (uint256 i = 0; i < diamonds.length; i++) {
+      string memory diamondName = diamonds[i].diamond;
+      address diamondAddress;
+      FacetCut[] memory newCuts;
+      string[] memory facetNames = new string[](diamonds[i].numFacets);
+      address[] memory facetAddresses = new address[](diamonds[i].numFacets);
+
+      for (uint256 j = 0; j < diamonds[i].numFacets; j++) {
+        facetAddresses[j] = diamonds[i].facets[j].deployedAddress;
+        facetNames[j] = diamonds[i].facets[j].facetName;
+      }
+
+      bytes32 diamondNameHash = keccak256(abi.encodePacked(diamondName));
+
+      if (diamondNameHash == keccak256(abi.encodePacked("space"))) {
+        // deploy space diamond by facets
+        diamondAddress = getDeployment("space");
+        removeRemoteFacetsByAddresses(deployer, diamondAddress, facetAddresses);
+        deploySpace.diamondInitParamsFromFacets(deployer, facetNames);
```

Comment:
deploySpace.addCut(facetName)

Reply:
addCut is called for each facet in facetNames within the body of diamondInitParamsFromFacets